### PR TITLE
Feature: UI Kit - Buttons

### PR DESCRIPTION
**Pull request by akueisara**
## Summary
- [x] Enable [viewBinding](https://developer.android.com/topic/libraries/view-binding) for the libsource module and the Kotlin sample project
- [x] Add [navigation libraries](https://developer.android.com/guide/navigation/navigation-getting-started) for the Kotlin sample project (multiple fragments in one activity using navigation)
- [x] Create a custom VirtusizeButton in the ui module to implement different types of the buttons
(Reference: https://design-system.virtusize.com/?path=/story/button--story)

## Demo
May-18-2021 16-20-14

### CODE REVIEW

File: libsource/build.gradle [hunk 1]
```diff
@@ -31,6 +30,10 @@ android {
             returnDefaultValues = true
         }
     }
+
+    buildFeatures {
+        viewBinding = true
```
Comment (akueisara) [inline libsource/build.gradle:35]: Enable viewBinding

File: sampleAppKotlin/build.gradle [hunk 2]
```diff
@@ -1,6 +1,9 @@
-apply plugin: 'com.android.application'
-apply plugin: 'kotlin-android'
-apply plugin: 'kotlin-android-extensions'
+plugins {
+    id 'com.android.application'
+    id 'kotlin-android'
+    id 'kotlin-kapt'
+    id 'androidx.navigation.safeargs.kotlin'
```
Comment (akueisara) [inline sampleAppKotlin/build.gradle:5]: Add navigation libraries

File: sampleAppKotlin/build.gradle [hunk 3]
```diff
@@ -26,14 +29,33 @@ android {
...
+        sourceCompatibility JavaVersion.VERSION_1_8
+        targetCompatibility JavaVersion.VERSION_1_8
+    }
+
+    kotlinOptions {
+        jvmTarget = '1.8'
+    }
+
+    buildFeatures {
+        viewBinding = true
```
Comment (akueisara) [inline sampleAppKotlin/build.gradle:43]: Enable viewBinding

File: sampleAppKotlin/build.gradle [hunk 4]
```diff
@@ -26,14 +29,33 @@ android {
...
+
     implementation"org.jetbrains.kotlin:kotlin-stdlib-jdk7:$kotlin_version"
+    implementation "androidx.core:core-ktx:$core_ktx_version"
     implementation "androidx.appcompat:appcompat:$appcompat_version"
     implementation "androidx.constraintlayout:constraintlayout:$constraint_layout_version"
-
-    implementation 'com.github.virtusize:integration_android:2.1.4'
-//    implementation project(":libsource")
+    implementation "androidx.navigation:navigation-fragment-ktx:$nav_version"
+    implementation "androidx.navigation:navigation-ui-ktx:$nav_version"
```
Comment (akueisara) [inline sampleAppKotlin/build.gradle:59]: Add navigation libraries

File: sampleAppKotlin/src/main/res/navigation/nav_graph.xml [hunk 5]
```diff
@@ -0,0 +1,29 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (akueisara) [inline sampleAppKotlin/src/main/res/navigation/nav_graph.xml:1]: This is the navigation graph.
It looks like the screenshot on Android Studio
Screen Shot 2021-05-18 at 16 14 01

File: ui/build.gradle [hunk 6]
```diff
@@ -23,6 +23,12 @@ android {
             consumerProguardFiles 'proguard-rules.pro'
         }
     }
+
+    sourceSets {
+        main {
+            res.srcDirs = ['src/main/res', 'src/main/java/com/virtusize/ui/button/res']
```
Comment (akueisara) [inline ui/build.gradle:29]: Create resource directories under different folders

File: ui/src/main/java/com/virtusize/ui/button/VirtusizeButtonTextSize.kt [hunk 7]
```diff
@@ -0,0 +1,8 @@
+package com.virtusize.ui.button
+
+/**
+ * This enum contains the text sizes that can be applied to the Virtusize Button
+ */
+enum class VirtusizeButtonTextSize {
+    DEFAULT, SMALLER, NORMAL, LARGE, LARGER
```
Comment (gpark-vs) [inline ui/src/main/java/com/virtusize/ui/button/VirtusizeButtonTextSize.kt:7]: So misc things..Might it be this?
```
DEFAULT,
SMALLER,
NORMAL,
LARGE,
LARGER
```

File: ui/src/main/java/com/virtusize/ui/button/VirtusizeButtonSize.kt [hunk 8]
```diff
@@ -0,0 +1,8 @@
+package com.virtusize.ui.button
+
+/**
+ * This enum contains the sizes that can be applied to the Virtusize Button
+ */
+enum class VirtusizeButtonSize {
+    STANDARD, SMALL
```
Comment (gpark-vs) [inline ui/src/main/java/com/virtusize/ui/button/VirtusizeButtonSize.kt:7]: So misc things..Might it be this?
```
STANDARD,
SMALL
```

File: sampleAppKotlin/src/main/res/values/styles.xml [hunk 9]
```diff
@@ -8,4 +8,15 @@
         <item name="colorAccent">@color/colorAccent</item>
     </style>
 
+    <style name="VirtusizeDesignSystemFlow">
+        <item name="android:paddingStart">@dimen/virtusize_16dp</item>
+        <item name="android:paddingEnd">@dimen/virtusize_16dp</item>
+        <item name="flow_horizontalBias">0</item>
```
Comment (gpark-vs) [inline sampleAppKotlin/src/main/res/values/styles.xml:14]: Might it be `flowHorizontalBias `?

Comment (akueisara) [reply sampleAppKotlin/src/main/res/values/styles.xml:14]: This is a default attribute from the official Android library

File: ui/src/main/java/com/virtusize/ui/button/VirtusizeButtonSize.kt (hunk 8 above)
Comment (akueisara) [reply ui/src/main/java/com/virtusize/ui/button/VirtusizeButtonSize.kt]: Yes, I've made the change

File: ui/src/main/java/com/virtusize/ui/button/VirtusizeButtonTextSize.kt (hunk 7 above)
Comment (akueisara) [reply ui/src/main/java/com/virtusize/ui/button/VirtusizeButtonTextSize.kt]: Yes, I've made the change

File: libsource/src/main/res/values/dimens.xml [hunk 10]
```diff
@@ -1,9 +1,5 @@
 <?xml version="1.0" encoding="utf-8"?>
 <resources>
-    <dimen name="virtusizeDp8">8dp</dimen>
-    <dimen name="virtusizeDp14">14dp</dimen>
-    <dimen name="virtusizeDp16">16dp</dimen>
-
     <dimen name="virtusize_button_textSize">14sp</dimen>
     <dimen name="virtusize_inpage_mini_message_textSize">14sp</dimen>
     <dimen name="virtusize_inpage_default_textSize">12sp</dimen>
```
Comment (gpark-vs) [inline libsource/src/main/res/values/dimens.xml:5]: Might it be text_size..? 🤔

Comment (akueisara) [reply libsource/src/main/res/values/dimens.xml]: yeah, sounds good. I've made the change. Thanks for spotting it!

**Review by gpark-vs — APPROVED** [review]
LGTM 👍 
Thanks for all your following up!

**Pull request merged**